apps/client/src/app/core/scheme/criteria/criteria.component.ts

```diff
@@ -0,0 +1,211 @@
+  }
+  errors = {
+    subIndex: '',
+    name: '',
+    alias: '',
+    requiresEvidence: '',
+    helpText: ''
+  }
+  @Input() criterion: Criteria[] = []
+  @Input() indicatorIndex: number = 0
+  visibleCreate = false
+  visibleEdit = false
+  fetchCreateCriterion: boolean = false
+  fetchEditCriterion: boolean = false
+  criteriaEdit: formPayloadEdit = {
+    id: 0,
+    subIndex: 0,
+    name: '',
+    alias: '',
+    requiresEvidence: false,
+    helpText: ''
+  }
+  closeDialog() {
+    this.visibleCreate = false
+    this.visibleEdit = false
+    this.formGroup.reset()
+    this.formGroup.controls.subIndex.setValue(1)
+    this.formGroup.controls.requiresEvidence.setValue(true)
+    this.errors = {
+      subIndex: '',
+      name: '',
+      alias: '',
+      requiresEvidence: '',
+      helpText: ''
+    }
+  }
+  showDialogEdit(criteria: Criteria) {
+    this.visibleEdit = true
+    this.criteriaEdit = criteria
+    this.formGroup.controls.subIndex.setValue(this.criteriaEdit.subIndex)
+    this.formGroup.controls.name.setValue(this.criteriaEdit.name)
+    this.formGroup.controls.alias.setValue(this.criteriaEdit.alias)
+    this.formGroup.controls.requiresEvidence.setValue(
+      this.criteriaEdit.requiresEvidence
+    )
+    this.formGroup.controls.helpText.setValue(this.criteriaEdit.helpText)
+  }
+  confirmationDelete(event: Event, id: number, name: string) {
+    this.confirmationService.confirm({
+      target: event.target as EventTarget,
+      message: `¿Estás seguro de que quieres eliminar el criterio <strong>${name}</strong>?`,
+      header: 'Eliminar criterio',
+      icon: 'pi pi-info-circle',
+      acceptButtonStyleClass: 'p-button-danger p-button-text',
+      rejectButtonStyleClass: 'p-button-text p-button-text',
+      acceptIcon: 'none',
+      rejectIcon: 'none',
+      acceptLabel: 'Sí',
+      rejectLabel: 'No',
+
+      accept: () => {
+        this.toast.show('info', 'Eliminando..', 'Eliminando criterio..')
+        this.onDelete(id)
+      },
+      reject: () => {
+        this.toast.show('error', 'Rechazado', 'Haz rechazado la eliminación')
+      }
+    })
+  }
+  getCriterionPerIndicator(indicatorId: number) {
+    this.criteriaService
+      .getAllCriterionByIndicatorIndex(indicatorId)
+      .subscribe({
+        next: (res) => {
+          this.criterion = res.status === 'success' ? res.data : this.criterion
+        },
+        error: (e) => {
+          console.error(e)
+          this.toast.show('error', 'Error', e.error.data.message)
+        }
+      })
+  }
+  onDelete(id: number) {
+    this.criteriaService.delete(id).subscribe({
+      next: () => {
+        this.toast.show('success', 'Eliminado', 'Criterio eliminado con éxito')
+        this.getCriterionPerIndicator(this.indicatorIndex)
+      },
+      error: (e) => {
+        console.error(e)
+        this.toast.show('error', 'Error', e.error.data.message)
+      }
+    })
+  }
+  onCreate() {
```

Comment:
No estás chequeando que el form sea válido

```suggestion
  onCreate() {
    if (this.formGroup.invalid) return
    ...
```